src/agent/onefuzz-task/src/tasks/coverage/generic.rs

```diff
@@ -457,49 +457,78 @@ impl<'a> TaskContext<'a> {
         Ok(())
     }
 
-    pub async fn save_and_sync_coverage(&self) -> Result<()> {
-        // JSON binary coverage.
-        let binary = self.coverage.clone();
-        let json = BinaryCoverageJson::V1(BinaryCoverageJsonV1::from(binary));
-        let text = serde_json::to_string(&json).context("serializing binary coverage")?;
-        let path = self.config.coverage.local_path.join(COVERAGE_FILE);
-        fs::write(&path, &text)
-            .await
-            .with_context(|| format!("writing coverage to {}", path.display()))?;
-
-        // JSON source coverage.
-        let source = self.source_coverage().await?;
-        let json = SourceCoverageJson::V1(SourceCoverageJsonV1::from(source.clone()));
-        let text = serde_json::to_string(&json).context("serializing source coverage")?;
-        let path = self.config.coverage.local_path.join(SOURCE_COVERAGE_FILE);
-        fs::write(&path, &text)
-            .await
-            .with_context(|| format!("writing source coverage to {}", path.display()))?;
+    pub async fn save_coverage(
+        coverage: &BinaryCoverage,
+        source_allowlist: &AllowList,
+        binary_coverage_path: &Path,
+        source_coverage_path: &Path,
+        copbertura_file_path: &Path,
+    ) -> Result<()> {
+        Self::save_binary_coverage(coverage, binary_coverage_path)?;
+
+        let source = binary_to_source_coverage(coverage, source_allowlist.clone())?;
```

Comment:
We used to do this in a `spawn_blocking` call, was that intentionally removed? I think we're still doing the same amount of blocking I/O as before